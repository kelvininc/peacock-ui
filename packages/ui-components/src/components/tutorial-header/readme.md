# kv-tutorial-header



<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- Required properties only -->
<kv-tutorial-header label="Label 1" description="Example label" />

<!-- Custom separator -->
<kv-tutorial-header label="Label 1" description="Example label" separator="/" />
```


### React

```jsx
import React from 'react';

import { KvTutorialHeader } from '@kelvininc/react-ui-components';

export const TutorialHeaderExamples = () => (
	<>
		{/*-- Required properties only --*/}
		<KvTutorialHeader label="Label 1" description="Example label" />

		{/*-- Custom separator --*/}
		<KvTutorialHeader label="Label 1" description="Example label" separator="/" />
	</>
);
```



## Properties

| Property                   | Attribute     | Description                                                                 | Type     | Default     |
| -------------------------- | ------------- | --------------------------------------------------------------------------- | -------- | ----------- |
| `description` _(required)_ | `description` | (required) A description of the state (e.g the description of a step)       | `string` | `undefined` |
| `label` _(required)_       | `label`       | (required) A title to describe a state (e.g step)                           | `string` | `undefined` |
| `separator`                | `separator`   | (optional) A separator character to place between the label and description | `string` | `'-'`       |


## CSS Custom Properties

| Name           | Description                   |
| -------------- | ----------------------------- |
| `--text-color` | The color to use for all text |


----------------------------------------------


