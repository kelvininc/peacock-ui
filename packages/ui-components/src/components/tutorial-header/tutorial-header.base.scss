@import '../../assets/styles/globals';

/**
 * @prop --text-color: The color to use for all text
 */

.tutorial-header-container {
	display: flex;
	align-items: center;
	padding: $spacing-4x;

	.label,
	.description,
	.separator {
		color: var(--text-color);
	}

	.label,
	.separator {
		@include kv-font-span-semibold;
	}

	.description {
		@include ellipsis;
		@include kv-font-span-regular;
	}

	.separator {
		padding: 0 $spacing;
	}
}
