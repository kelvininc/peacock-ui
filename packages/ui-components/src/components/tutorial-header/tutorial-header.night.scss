@import './tutorial-header.base';

:host {
	--text-color: #{kv-color('neutral-3')};
}
