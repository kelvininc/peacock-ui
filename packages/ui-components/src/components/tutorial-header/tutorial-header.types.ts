export interface ITutorialHeader {
	/** (required) A title to describe a state (e.g step) */
	label: string;
	/** (required) A description of the state (e.g the description of a step) */
	description: string;
	/** (optional) A separator character to place between the label and description */
	separator: string;
}
