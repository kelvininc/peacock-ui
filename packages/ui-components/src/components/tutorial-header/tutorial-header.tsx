import { Component, Host, Prop, h } from '@stencil/core';
import { ITutorialHeader } from './tutorial-header.types';

@Component({
	tag: 'kv-tutorial-header',
	styleUrls: {
		night: 'tutorial-header.night.scss'
	},
	shadow: true
})
export class KvTutorialHeader implements ITutorialHeader {
	/** @inheritdoc */
	@Prop({ reflect: true }) label!: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) description!: string;
	/** @inheritdoc */
	@Prop({ reflect: true }) separator: string = '-';

	render() {
		return (
			<Host>
				<div class="tutorial-header-container">
					<span class="label">{this.label}</span>
					<span class="separator">{this.separator}</span>
					<span class="description">{this.description}</span>
				</div>
			</Host>
		);
	}
}
