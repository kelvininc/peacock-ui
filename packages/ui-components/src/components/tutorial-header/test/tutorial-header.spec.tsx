import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { KvTutorialHeader } from '../tutorial-header';

describe('TutorialHeader (unit tests)', () => {
	let page: SpecPage;
	let comp: KvTutorialHeader;

	describe('when rendering with defaults', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvTutorialHeader],
				html: `<kv-tutorial-header label="Step 1" description="Choose an application" />`
			});
			comp = page.rootInstance;
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should have the correct label assigned', () => {
			expect(comp.label).toEqual('Step 1');
		});

		it('should have the correct description assigned', () => {
			expect(comp.description).toEqual('Choose an application');
		});

		it('should have the default separator assigned', () => {
			expect(comp.separator).toEqual('-');
		});
	});

	describe('when rendering with a different separator', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvTutorialHeader],
				html: `<kv-tutorial-header label="Step 1" description="Choose an application" separator="⬤" />`
			});
			comp = page.rootInstance;
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should have the correct separator', () => {
			expect(comp.separator).toEqual('⬤');
		});
	});
});
