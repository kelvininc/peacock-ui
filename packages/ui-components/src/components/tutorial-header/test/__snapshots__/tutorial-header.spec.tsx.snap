// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialHeader (unit tests) when rendering with a different separator should match the snapshot 1`] = `
<kv-tutorial-header description="Choose an application" label="Step 1" separator="⬤">
  <mock:shadow-root>
    <div class="tutorial-header-container">
      <span class="label">
        Step 1
      </span>
      <span class="separator">
        ⬤
      </span>
      <span class="description">
        Choose an application
      </span>
    </div>
  </mock:shadow-root>
</kv-tutorial-header>
`;

exports[`TutorialHeader (unit tests) when rendering with defaults should match the snapshot 1`] = `
<kv-tutorial-header description="Choose an application" label="Step 1" separator="-">
  <mock:shadow-root>
    <div class="tutorial-header-container">
      <span class="label">
        Step 1
      </span>
      <span class="separator">
        -
      </span>
      <span class="description">
        Choose an application
      </span>
    </div>
  </mock:shadow-root>
</kv-tutorial-header>
`;
